src/main/java/com/strandls/user/controller/UserController.java

```diff
@@ -175,6 +175,35 @@ public Response updateUserEmailPreferences(@Context HttpServletRequest request,
 		return Response.status(Status.OK).entity(user).build();
 	}
 
+	@GET
+	@Path(ApiConstants.UNSUBSCRIBE + "/{token}")
```

Comment:
check if you can remove the token from the path param and put it in the request header itself, as the user is anyway going to unsubscribe ownself.